cloudregister/registerutils.py

```diff
@@ -346,30 +346,51 @@ def enable_repository(repo_name):
     """Enable the given repository"""
 
     cmd = ['zypper', 'mr', '-e', repo_name]
-    res = exec_subprocess(cmd)
-    if res:
-        log.error('Unable to enable repository %s' % repo_name)
+    _, _, failed = exec_subprocess(cmd)
+    if failed:
+        log.error(
+            'Unable to enable repository {}'.format(repo_name)
+        )
 
 
 # ----------------------------------------------------------------------------
-def exec_subprocess(cmd, return_output=False):
-    """Execute the given command as a subprocess (blocking)
-       Returns one off:
-           - exit code of the command
-           - stdout, stderr and exit code
-           - -1 indicates an exception"""
+def exec_subprocess(cmd, tolog=True, disconnect=False):
```

Comment:
We must obfuscate the registration code. With this change an invocation of `registercloudguest -r 123456` would end up logging `SUSEConnect --url....... -r 123456` and that is not an option. So we eithre have to disable logging, i.e. set `tolog` false when we get a registration code from the user or we need to put a filter in this code that turns the reg code into `XXXXXX` or some other meaningless collection of characters.

Reply:
I added a commit to prevent the exec logging from the exec_subprocess in register_product(). If you look into this method you also see that we create a special obfuscated log information for the call. Thus there is no need to let exec_subprocess log it again. That should do the trick